ext/rbtrace.c

```diff
@@ -803,17 +804,37 @@ rbtracer_watch(uint32_t threshold, bool cpu_time)
   }
 }
 
+static int
+get_msgq_key(pid_t pid)
+{
+#ifndef __linux__
+  return(pid);
+#else
+  char buffer[128];
+  int status;
+  struct stat buf;
+  snprintf(buffer, 128, "/proc/%d", pid);
+  memset(&buf, 0, sizeof(buf));
+  status = stat(buffer, &buf);
+  if (status == 0) {
+    return(buf.st_ino);
+  }
+  return(pid);
```

Comment:
arguably, I could just fail here. But it is sort of unlikely that the stat in native land will fail and the stat in ruby land works, and they both return pid on error.